### PR TITLE
Error handling for invalid document

### PR DESCRIPTION
**Pull request by shinyichen**
Redirect to 404 page when document is not found

### CODE REVIEW

File: src/pages/abs/[id]/similar.tsx [hunk 1]
```diff
@@ -55,14 +57,16 @@ export const getServerSideProps: GetServerSideProps<ICitationsPageProps> = async
...
-  }
-
-  return {
-    props: {
-      docs: result.value.docs,
-      originalDoc,
-    },
-  };
+  return !originalDoc
+    ? { props: { error: 'Document not found' } }
```
Comment (thostetler) [inline src/pages/abs/[id]/similar.tsx:61]: Looks like we can shortcut this a bit by doing this: https://nextjs.org/blog/next-10#notfound-support

`{ notFound: true }` <-- will automatically show the 404 error page

Comment (thostetler) [reply src/pages/abs/[id]/similar.tsx:61]: We'll want to check on what kind of error we're getting and act accordingly

**Review by thostetler — COMMENTED** [review]
LGTM -- Take a look at https://nextjs.org/blog/next-10#notfound-support, and maybe we can make this error handling a little cleaner.

**Comment by shinyichen** [comment]
@thostetler Now separating the different errors, please see if this enough. Thanks.

**Pull request merged**